server/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -468,18 +480,39 @@ public void syncTranslog() throws IOException {
     }
 
     @Override
-    public Closeable acquireTranslogRetentionLock() {
-        return getTranslog().acquireRetentionLock();
+    public Translog.Snapshot newTranslogSnapshotBetween(long minSeqNo, long maxSeqNo) throws IOException {
```

Comment:
I think we can remove this as a follow up, in favor of readHistoryOperations ?

Reply:
Yes, we will remove this method after making primary-replica resync use Lucene.